pennylane/data/dataset.py

```diff
@@ -23,25 +23,34 @@
 
 class Dataset(ABC):
     "The dataset data type. Allows users to create datasets with their own data"
+    CFG_ATTRS = {"dtype", "folder", "file_prefix", "file_fstring", "_attributes", "__doc__"}
 
-    def __init__(self, dfile=None, dtype=None, **kwargs):
-        self.dfile = dfile
+    def __init__(self, dtype, folder, file_prefix, attributes=None, **kwargs):
```

Comment:
Should the keywords have default arguments for users to work with custom data (i.e. the one they have locally)?